source/digits_hits/include/GateJPETGammaPolarizationActor.hh

```diff
@@ -0,0 +1,302 @@
+	*/
+	void SetAllPhiAsNoNegative(G4bool setAsPlus);
+
+	/** Set generation of histogram as (dis)enable.
+	 * By default generation is enable;
+	 * @param: enable - set false to disable histograms generations
+	*/
+	void SetHistogramsEnable(G4bool enable);
+
+	/** Set generation of tests branch (for polarization) as (dis)enable.
+	 * By default generation is enable;
+	 * @param: enable - set false to disable tests branch generations
+	*/
+	void SetTestsBranchEnable(G4bool enable);
+
+	/** Set angle precision per bin in histograms - this mean that each bin has width equal angle_precison
+	 * By default precision is 2 degree.
+	 * @param: angle_precision - bin width (in degrees)
+	 * @exceptions: If value is to huge, what mean angle_precision >= 180.0 deg then this value is ignored.
+	*/
+	void SetAnglePrecisionPerBinInHistograms(G4double angle_precision);
+
+	/** Set (dis)enable saving data ony for desired number of particles per event which scattered.
+	 * The purpose of this to reduce number of data generated during simulation.
+	 * @param: enable - set true to use this, false - otherwise
+	 * */
+	void SetSaveOnlyWhenTheDesiredNumberOfParticlesHasScatteredFromEventEnable(G4bool enable);
+
+	/** Set desired number of particles scattered per event.
+	 * @param: particles_number - if number of scattered per event is equal particles_number then data will be saved
+	 * */
+	void SetDesiredNumberOfParticlesScatteredPerEvent(G4int particles_number);
+
+private: //Functions
+	/** Function generate perpendicular vector to vector 'a'
+	 * @param: a - vector for which we create perpendicular one
+	 * @return: perpendicular vector to 'a'
+	*/
+	G4ThreeVector SetPerpendicularVector(G4ThreeVector& a);
+
+	/* Special function just for this class.
+	* Extract values setted by Set functions.
+	* */
+	void StandardExtractFunction(const G4Step *step);
+
+	/** Function generate dsigma/dOmega as function of theta.
+	 * For this this function generate histogram which is histogram dsigma/dOgema divided by sin(theta) histogram.
+	 * */
+	void GenerateRealCrossSectionOfTheta();
+
+	/** Function normalize major of histograms
+	 * */
+	void NormalizeHistograms();
+
+	/** Function save numerical data to ASCII file
+	 * */
+	void SaveToFile(G4int eventID, G4int trackID, G4double phi, G4double theta, G4double polarization, G4ThreeVector interactionPoint, G4double totalEnergy, G4ThreeVector emisionPoint, G4double primeEnergy, G4double emisionPhi, G4double emisionTheta);
+
+	/** Function calculate phi and theta for scattered gamma.
+	 * @param: k0 - prime gamma momentum direction
+	 * @param: k - scattered gamma momentum direction
+	 * @param: e0 - prime gamma polarization
+	 * @param: theta - angle between k0 and k
+	 * @param: phi - angle between e0 and k projection on plane created by e0 and e0xk0
+	 * */
+	void GetThetaAndPhi(const G4ThreeVector k0, const G4ThreeVector k, const G4ThreeVector e0, G4double& theta, G4double& phi);
```

Comment:
Why here you dont pass by const reference?

Reply:
Becouse method **GetThetaAndPhi** modifies variables **theta** and **phi** - you can't modifie const reference objects in function.